src/server/pps/server/poller.go

```diff
@@ -282,20 +284,27 @@ func (m *ppsMaster) watchPipelines(ctx context.Context) {
 			if err != nil {
 				return errors.Wrap(err, "bad watch event key")
 			}
+			var e *pipelineEvent
 			switch event.Type {
 			case watch.EventPut:
-				m.eventCh <- &pipelineEvent{
+				e = &pipelineEvent{
 					eventType: writeEv,
 					pipeline:  pipelineName,
 					timestamp: time.Unix(event.Rev, 0),
 				}
 			case watch.EventDelete:
-				m.eventCh <- &pipelineEvent{
+				e = &pipelineEvent{
 					eventType: deleteEv,
 					pipeline:  pipelineName,
 					timestamp: time.Unix(event.Rev, 0),
 				}
 			}
+			select {
+			case m.eventCh <- e:
+				continue
+			case <-m.masterCtx.Done():
```

Comment:
The semantic change in this file: this ensures pollPipelines can exit after the PPS master restarts or loses mastership or something.